src/SoapCore/ServiceBodyWriter.cs

```diff
@@ -20,151 +22,173 @@ public class ServiceBodyWriter : BodyWriter
 		private readonly object _result;
 		private readonly Dictionary<string, object> _outResults;
 
-		public ServiceBodyWriter(SoapSerializer serializer, OperationDescription operation, string resultName, object result, Dictionary<string, object> outResults) : base(isBuffered: true)
```

Comment:
I followed this parameter and noted it was just lifted out of operation.ReturnName. I have pulled that logic in here and removed the extra parameter.